crud-backend/src/index.tsx

```diff
@@ -0,0 +1,60 @@
+import express, { Express, Request, Response, Application } from 'express';``
+// const express = require("express");
+import dotenv from 'dotenv';
+// const dotenv = require("dotenv");
+import { createClient } from '@supabase/supabase-js';
+// const createClient = require("@supabase/supabase-js")
+// const cors = require("cors");
+import cors from "cors";
+
+
+// for env file
+dotenv.config();
+
+const supabaseUrl = process.env.SUPABASE_URL ?? '';
+const supabasePublicAnonKey = process.env.SUPABASE_PUBLIC_ANON_KEY ?? '';
+
+console.log(supabaseUrl);
+
+const supabase = createClient(supabaseUrl, supabasePublicAnonKey, {
+  auth: {
+    autoRefreshToken: false,
+    persistSession: true,
+    detectSessionInUrl: false
+  }
+});
```

Comment:
It's useful to put this behind a helper function like `createSupabase`. You don't need to access the variables `supabaseUrl` and `supabasePublicAnonKey` elsewhere, so that gives you better scoping and better abstracts the code.